abi/decode.go

```diff
@@ -153,6 +159,9 @@ func readInteger(t *Type, b []byte) interface{} {
 
 func readFunctionType(t *Type, word []byte) ([24]byte, error) {
 	res := [24]byte{}
+	if len(word) < 32 {
```

Comment:
This check is not necessary since `decode` already checks that the size is larger than 32 bytes.